src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -263,6 +272,62 @@ public BulkInsertOperation(string database, IDocumentStore store, BulkInsertOpti
 
             _generateEntityIdOnTheClient = new GenerateEntityIdOnTheClient(_requestExecutor.Conventions,
                 entity => AsyncHelpers.RunSync(() => _requestExecutor.Conventions.GenerateDocumentIdAsync(database, entity)));
+
+            _streamLock = new AsyncReaderWriterLock();
+            _lastWriteToStream = DateTime.UtcNow;
+            _timer = new Timer(HandleHeartbeat,
+                new WeakReference(this),
+                HeartbeatCheckInterval,
+                HeartbeatCheckInterval);
+        }
+
+        private static void HandleHeartbeat(object state)
+        {
+            var bulkInsert = (BulkInsertOperation)(((WeakReference)state).Target);
```

Comment:
safe cast for state and then use TryGetTarget?

Reply:
done